### PR TITLE
Highlight selection fix

### PR DESCRIPTION
**Pull request by marredal**
Added highlight selection.
Also conventiently fixed mouse selection from working when holding down the button
(Before: Pressed, Now: Triggered).

### CODE REVIEW

File: src/Video/Renderer.cpp [hunk 1]
```diff
@@ -110,8 +110,8 @@ void Renderer::PrepareStaticMeshRendering(const glm::mat4& viewMatrix, const glm
     staticRenderProgram->PreRender(viewMatrix, projectionMatrix);
 }
 
-void Renderer::RenderStaticMesh(Geometry::Geometry3D* geometry, const Texture2D* albedo, const Texture2D* normal, const Texture2D* metallic, const Texture2D* roughness, const glm::mat4 modelMatrix) {
-    staticRenderProgram->Render(geometry, albedo, normal, metallic, roughness, modelMatrix);
+void Renderer::RenderStaticMesh(Geometry::Geometry3D* geometry, const Texture2D* albedo, const Texture2D* normal, const Texture2D* metallic, const Texture2D* roughness, const glm::mat4 modelMatrix, bool isSelected) {
+    staticRenderProgram->Render(geometry, albedo, normal, metallic, roughness, modelMatrix, isSelected);
 }
 
 void Renderer::AntiAlias(RenderSurface* renderSurface) {
```
Comment (NinjaDanz3r) [inline src/Video/Renderer.cpp]: MINOR The function 'AntiAlias' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 20 issues
* ![MINOR][MINOR] 20 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [Mesh.cpp#L9](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/36c0a269a61d2ae498e53ec767b832dfa8de8f21/src/Engine/Component/Mesh.cpp#L9): Member variable 'Mesh::isSelected' is not initialized in the constructor. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AuninitMemberVar)
1. ![MINOR][MINOR] [RenderManager.cpp#L313](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/36c0a269a61d2ae498e53ec767b832dfa8de8f21/src/Engine/Manager/RenderManager.cpp#L313): Variable 'name' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [RenderManager.cpp#L323](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/36c0a269a61d2ae498e53ec767b832dfa8de8f21/src/Engine/Manager/RenderManager.cpp#L323): The function 'GetAnimations' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L341](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/36c0a269a61d2ae498e53ec767b832dfa8de8f21/src/Engine/Manager/RenderManager.cpp#L341): The function 'GetDirectionalLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L380](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/36c0a269a61d2ae498e53ec767b832dfa8de8f21/src/Engine/Manager/RenderManager.cpp#L380): The function 'GetMaterials' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L398](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/36c0a269a61d2ae498e53ec767b832dfa8de8f21/src/Engine/Manager/RenderManager.cpp#L398): The function 'GetMeshes' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L418](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/36c0a269a61d2ae498e53ec767b832dfa8de8f21/src/Engine/Manager/RenderManager.cpp#L418): The function 'GetPointLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L439](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/36c0a269a61d2ae498e53ec767b832dfa8de8f21/src/Engine/Manager/RenderManager.cpp#L439): The function 'GetSpotLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [Renderer.cpp#L27](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/36c0a269a61d2ae498e53ec767b832dfa8de8f21/src/Video/Renderer.cpp#L27): Member variable 'Renderer::skinRenderProgram' is not initialized in the constructor. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AuninitMemberVar)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 20 issues
* ![MINOR][MINOR] 20 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [Mesh.cpp#L9](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/2d7942d7cece280fe00b25065898eb4987d6e776/src/Engine/Component/Mesh.cpp#L9): Member variable 'Mesh::isSelected' is not initialized in the constructor. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AuninitMemberVar)
1. ![MINOR][MINOR] [RenderManager.cpp#L366](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/2d7942d7cece280fe00b25065898eb4987d6e776/src/Engine/Manager/RenderManager.cpp#L366): Variable 'name' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [RenderManager.cpp#L376](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/2d7942d7cece280fe00b25065898eb4987d6e776/src/Engine/Manager/RenderManager.cpp#L376): The function 'GetAnimations' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L394](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/2d7942d7cece280fe00b25065898eb4987d6e776/src/Engine/Manager/RenderManager.cpp#L394): The function 'GetDirectionalLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L433](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/2d7942d7cece280fe00b25065898eb4987d6e776/src/Engine/Manager/RenderManager.cpp#L433): The function 'GetMaterials' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L451](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/2d7942d7cece280fe00b25065898eb4987d6e776/src/Engine/Manager/RenderManager.cpp#L451): The function 'GetMeshes' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L471](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/2d7942d7cece280fe00b25065898eb4987d6e776/src/Engine/Manager/RenderManager.cpp#L471): The function 'GetPointLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L492](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/2d7942d7cece280fe00b25065898eb4987d6e776/src/Engine/Manager/RenderManager.cpp#L492): The function 'GetSpotLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [Renderer.cpp#L27](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/2d7942d7cece280fe00b25065898eb4987d6e776/src/Video/Renderer.cpp#L27): Member variable 'Renderer::skinRenderProgram' is not initialized in the constructor. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AuninitMemberVar)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/Engine/Component/Mesh.hpp [hunk 2]
```diff
@@ -30,5 +30,21 @@ namespace Component {
              */
             Video::Geometry::Geometry3D* geometry = nullptr;
 
+            /// Highlight selection.
```
Comment (SwedishSock) [inline src/Engine/Component/Mesh.hpp]: Both this and SetSelected has /// comment "Highlight Selection"

File: src/Engine/Component/Mesh.hpp [hunk 3]
```diff
@@ -30,5 +30,21 @@ namespace Component {
...
+            /**
+            * @return True if selected else false.
```
Comment (SwedishSock) [inline src/Engine/Component/Mesh.hpp]: Maybe a better description would be "@return whether it is selected or not" ?

File: src/Engine/Component/Mesh.hpp [hunk 4]
```diff
@@ -30,5 +30,21 @@ namespace Component {
...
+            */
+            bool GetSelected() const;
+
+            /// Highlight selection.
+            /**
+            * @param Set true if mesh is selected, else false.
```
Comment (SwedishSock) [inline src/Engine/Component/Mesh.hpp]: Incorrect doxygen, @param should be followed by parameter name (in this case value) followed by description.

File: src/Engine/Component/Mesh.hpp [hunk 5]
```diff
@@ -30,5 +30,21 @@ namespace Component {
...
+            */
+            void SetSelected(bool value);
+
+    private:
+        /// If mesh is selected in the editor.
```
Comment (SwedishSock) [inline src/Engine/Component/Mesh.hpp]: Private should not have doxygen comments, only regular ones.

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 20 issues
* ![MINOR][MINOR] 19 minor
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [RenderManager.cpp#L314](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/cf10bb86f94c6e3acadd010a1595913e0595c488/src/Engine/Manager/RenderManager.cpp#L314): Variable 'name' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [RenderManager.cpp#L324](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/cf10bb86f94c6e3acadd010a1595913e0595c488/src/Engine/Manager/RenderManager.cpp#L324): The function 'GetAnimations' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L342](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/cf10bb86f94c6e3acadd010a1595913e0595c488/src/Engine/Manager/RenderManager.cpp#L342): The function 'GetDirectionalLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L381](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/cf10bb86f94c6e3acadd010a1595913e0595c488/src/Engine/Manager/RenderManager.cpp#L381): The function 'GetMaterials' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L399](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/cf10bb86f94c6e3acadd010a1595913e0595c488/src/Engine/Manager/RenderManager.cpp#L399): The function 'GetMeshes' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L419](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/cf10bb86f94c6e3acadd010a1595913e0595c488/src/Engine/Manager/RenderManager.cpp#L419): The function 'GetPointLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L440](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/cf10bb86f94c6e3acadd010a1595913e0595c488/src/Engine/Manager/RenderManager.cpp#L440): The function 'GetSpotLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [Renderer.cpp#L27](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/cf10bb86f94c6e3acadd010a1595913e0595c488/src/Video/Renderer.cpp#L27): Member variable 'Renderer::skinRenderProgram' is not initialized in the constructor. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AuninitMemberVar)
1. ![MINOR][MINOR] [Renderer.cpp#L27](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/cf10bb86f94c6e3acadd010a1595913e0595c488/src/Video/Renderer.cpp#L27): Member variable 'Renderer::postProcessing' is not initialized in the constructor. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AuninitMemberVar)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 20 issues
* ![MINOR][MINOR] 19 minor
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [RenderManager.cpp#L314](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/f2fab9350c58bb645bb6a2a2dcfde4307f1cda0b/src/Engine/Manager/RenderManager.cpp#L314): Variable 'name' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [RenderManager.cpp#L324](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/f2fab9350c58bb645bb6a2a2dcfde4307f1cda0b/src/Engine/Manager/RenderManager.cpp#L324): The function 'GetAnimations' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L342](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/f2fab9350c58bb645bb6a2a2dcfde4307f1cda0b/src/Engine/Manager/RenderManager.cpp#L342): The function 'GetDirectionalLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L381](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/f2fab9350c58bb645bb6a2a2dcfde4307f1cda0b/src/Engine/Manager/RenderManager.cpp#L381): The function 'GetMaterials' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L399](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/f2fab9350c58bb645bb6a2a2dcfde4307f1cda0b/src/Engine/Manager/RenderManager.cpp#L399): The function 'GetMeshes' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L419](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/f2fab9350c58bb645bb6a2a2dcfde4307f1cda0b/src/Engine/Manager/RenderManager.cpp#L419): The function 'GetPointLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L440](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/f2fab9350c58bb645bb6a2a2dcfde4307f1cda0b/src/Engine/Manager/RenderManager.cpp#L440): The function 'GetSpotLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [Renderer.cpp#L27](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/f2fab9350c58bb645bb6a2a2dcfde4307f1cda0b/src/Video/Renderer.cpp#L27): Member variable 'Renderer::skinRenderProgram' is not initialized in the constructor. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AuninitMemberVar)
1. ![MINOR][MINOR] [Renderer.cpp#L27](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/f2fab9350c58bb645bb6a2a2dcfde4307f1cda0b/src/Video/Renderer.cpp#L27): Member variable 'Renderer::postProcessing' is not initialized in the constructor. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AuninitMemberVar)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**